src/game/boe.graphics.cpp

```diff
@@ -738,7 +738,7 @@ void draw_terrain(short	mode) {
 			where_draw.x += i - 6;
 			where_draw.y += j - 6;
 			if(!(is_out()))
-				light_area[i][j] = (is_town()) ? pt_in_light(view_loc,where_draw) : combat_pt_in_light(where_draw);
+				light_area[i][j] = (is_town()) ? (can_see_light(view_loc,where_draw,sight_obscurity)<=4) : combat_pt_in_light(where_draw);
```

Comment:
Hmm. Which of these is legacy behaviour in Blades of Exile? What about in Exile III? And what does this change actually look like?

Seeing explored regions (terrain only) through walls is expected behaviour, at least in towns with an inherent light level. I'm not sure if that applies to dark towns.

Reply:
I don't know why, but I keep seeing regions explored in cities with an inherent level of light.

Also note that after in the code we have :
```
			if(!(is_out()) && ((where_draw.x < 0) || (where_draw.x > univ.town->max_dim - 1)
								|| (where_draw.y < 0) || (where_draw.y > univ.town->max_dim - 1))) {
...
				if(can_see_light(view_loc,where_draw,sight_obscurity) < 5)
					can_draw = 1;
				else can_draw = 0;

```
so the use of can_see_light seems more natural (and when I made this change, I was probably correcting a problem in dark cities)

Reply:
I checked the original code and it does use `pt_in_light` here, so we'd have to decide whether that's a bug in the original game (and thus needs fixing) or a quirk of the original game (so no fix required).